rancher2/schema_global_dns_provider.go

```diff
@@ -0,0 +1,78 @@
+package rancher2
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// Schemas
+func globalDNSProviderFields() map[string]*schema.Schema {
+	s := map[string]*schema.Schema{
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"root_domain": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"dns_provider": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: false,
+			ValidateFunc: func(v interface{}, s string) (strings []string, errors []error) {
+				validDNSProviders := []string{"route53", "cloudflare", "alidns"}
+				provider := v.(string)
+				for _, p := range validDNSProviders {
+					if p == provider {
+						//strings = append(strings, provider)
+						return
+					}
+				}
+				errors = append(errors, fmt.Errorf("provider must be one of aws, cloudflare, alidns"))
+				return
+			},
```

Comment:
importing `"github.com/hashicorp/terraform-plugin-sdk/helper/validation"` module, you could use this ValidateFunc

```
ValidateFunc: validation.StringInSlice([]string{"route53", "cloudflare", "alidns"}, false),
```

Reply:
Thanks, I was trying to find this. I didn't see it in the TF docs.